### ISSUE TITLE
There are a bunch of pydoctor errors in twisted.web.iweb

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6196]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6196|
|Type|enhancement|
|Created|2012-11-28 14:05:21Z|
|Branch|https://github.com/twisted/twisted/tree/iweb-api-errors-6196|

```
twisted.web.iweb.IBodyProducer:372 invalid ref to IConsumer
twisted.web.iweb.IRequest.getHost:107 invalid ref to IAddress
twisted.web.iweb.IRequest.getClientIP:115 invalid ref to NoneType
twisted.web.iweb.IRequest.getClient:125 invalid ref to NoneType
twisted.web.iweb.IRequest.setLastModified:266 invalid ref to NOT_MODIFIED
twisted.web.iweb.IRequest.setLastModified:266 invalid ref to CACHED
twisted.web.iweb.IRequest.setETag:288 invalid ref to NOT_MODIFIED
twisted.web.iweb.IRequest.setETag:288 invalid ref to PRECONDITION_FAILED
twisted.web.iweb.IRequest.setETag:288 invalid ref to CACHED
twisted.web.iweb.ICredentialFactory:326 invalid ref to ICredentials
twisted.web.iweb.IBodyProducer:372 invalid ref to IConsumer
twisted.web.iweb.IBodyProducer:372 invalid ref to IConsumer
twisted.web.iweb.IBodyProducer:372 invalid ref to IConsumer.unregisterProducer
twisted.web.iweb.IBodyProducer:372 invalid ref to Deferred
twisted.web.iweb.IBodyProducer:372 invalid ref to IConsumer.write
twisted.web.iweb.IBodyProducer.length:399 invalid ref to UNKNOWN_LENGTH
twisted.web.iweb.IBodyProducer.startProducing:406 invalid ref to IConsumer
twisted.web.iweb.IBodyProducer.stopProducing:416 invalid ref to IProducer.stopProducing
twisted.web.iweb.IBodyProducer.stopProducing:416 invalid ref to Deferred
twisted.web.iweb.IBodyProducer.length:399 invalid ref to UNKNOWN_LENGTH
twisted.web.iweb.IBodyProducer.startProducing:406 invalid ref to IConsumer
twisted.web.iweb.IBodyProducer.startProducing:406 invalid ref to Deferred
twisted.web.iweb.IBodyProducer.startProducing:406 invalid ref to Failure
twisted.web.iweb.IBodyProducer.stopProducing:416 invalid ref to IProducer.stopProducing
twisted.web.iweb.IBodyProducer.stopProducing:416 invalid ref to Deferred
twisted.web.iweb.IRenderable.render:445 invalid ref to NoneType
twisted.web.iweb.IResponse.length:497 invalid ref to UNKNOWN_LENGTH
twisted.web.iweb.IResponse.deliverBody:505 invalid ref to IProtocol
twisted.web.iweb.IResponse.length:497 invalid ref to UNKNOWN_LENGTH
twisted.web.iweb.IResponse.deliverBody:505 invalid ref to IProtocol
```

Many of these are duplicated, but that's how pydoctor rolls.  Some of them also cannot be fixed presently, due to [links to module attributes not being supported](https://bugs.launchpad.net/pydoctor/+bug/685376).

<details><summary>Searchable metadata</summary>

```
trac-id__6196 6196
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_iweb_api_errors_6196 branches/iweb-api-errors-6196
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1354111521000000 1354111521000000
changetime__1354227661000000 1354227661000000
version__None None
owner__exarkun exarkun
cc__jknight cc__mwh
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [36463]) Branching to 'iweb-api-errors-6196'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_4"></a>|@mwhudson set owner to @exarkun|
|-|-|

Using r588 of pydoctor trunk, the invalid refs remaining in iweb are:

twisted.web.iweb.IRequest.setLastModified:261 invalid ref to http.NOT_MODIFIED
twisted.web.iweb.IRequest.setLastModified:261 invalid ref to http.CACHED
twisted.web.iweb.IRequest.setETag:283 invalid ref to http.NOT_MODIFIED
twisted.web.iweb.IRequest.setETag:283 invalid ref to http.CACHED

These seem to be problems with http's docstring: it says @var cached: not @var CACHED: and NOT_MODIFIED is not documented at all.  I think this should be fixed before the branch is merged.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [36467]) Spell `CACHED` correctly and also document `NOT_MODIFIED`.

refs [#6196](https://github.com/twisted/twisted/issues/6196)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun removed owner|
|-|-|

Thanks.  Some more errors fixed.

**Comment by twisted-trac** [comment]
|[<img alt="mwhudson's avatar" src="https://avatars.githubusercontent.com/u/672022?s=50" width="50" height="50">](https://github.com/mwhudson)<a name="note_8"></a>|@mwhudson set owner to @exarkun|
|-|-|

This looks good to merge now.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set status to `closed`|
|-|-|

(In [36471]) Merge iweb-api-errors-6196

Author: exarkun
Reviewer: mwhudson
Fixes: [#6196](https://github.com/twisted/twisted/issues/6196)

Fix a few API documentation issues in `twisted.web.iweb` and `twisted.web.http`.
